napari/layers/image/image.py

```diff
@@ -437,6 +440,15 @@ def rendering(self, rendering):
         self._rendering = Rendering(rendering)
         self.events.rendering()
 
+    @property
+    def loaded(self):
+        """Has the data for this layer been loaded yet.
+
+        With asynchronous loading the layer might exist but its data
+        for the current slice has not been loaded.
+        """
+        return self._slice is not None and self._slice.loaded
```

Comment:
For a long time in this PR `_slice` could be `None` but looks like that's no longer possible, which is really good, much simpler. So I moved that check. We'll find it out if it ever is`None` pretty quick because it will crash.